src/include/cell.i

```diff
@@ -11,12 +11,19 @@
  *     Check the value's validity window for sanity.
  */
 static inline void
-__cell_check_value_validity(WT_SESSION_IMPL *session, wt_timestamp_t start_ts, uint64_t start_txn,
+__cell_check_value_validity(WT_SESSION_IMPL *session, wt_timestamp_t durable_start_ts,
+  wt_timestamp_t durable_stop_ts, wt_timestamp_t start_ts, uint64_t start_txn,
   wt_timestamp_t stop_ts, uint64_t stop_txn)
 {
 #ifdef HAVE_DIAGNOSTIC
     char ts_string[2][WT_TS_INT_STRING_SIZE];
 
+    if (durable_start_ts > durable_stop_ts) {
+        __wt_errx(session, "a durable start timestamp %s newer than its durable stop timestamp %s",
+          __wt_timestamp_to_string(durable_start_ts, ts_string[0]),
+          __wt_timestamp_to_string(durable_stop_ts, ts_string[1]));
+        WT_ASSERT(session, durable_start_ts <= durable_stop_ts);
+    }
     if (start_ts != WT_TS_NONE && stop_ts == WT_TS_NONE) {
```

Comment:
I assume you can do this check for `durable_start_ts` and `durable_stop_ts` too?

Reply:
Added.